tmexp/create_bow.py

```diff
@@ -176,36 +183,39 @@ def create_bow(
     document_index = {}
     num_docs = 0
     with open(doc_output_path, "w", encoding="utf-8") as fout:
-        for doc in sorted(docs):
-            for i, refs in enumerate(docs[doc]):
-                doc_name = doc + SEP + str(i)
-                document_index[doc_name] = num_docs
-                num_docs += 1
-                fout.write(" ".join([doc_name] + refs) + "\n")
+        for repo in sorted(docs):
+            for doc in sorted(docs[repo]):
+                for i, refs in enumerate(sorted(docs[repo][doc])):
+                    doc_name = SEP.join((repo, doc, str(i)))
+                    document_index[doc_name] = num_docs
+                    num_docs += 1
+                    fout.write(" ".join([doc_name] + refs) + "\n")
     logger.info("Number of distinct documents : %d" % num_docs)
     logger.info("Saved document index in '%s'" % doc_output_path)
 
     logger.info("Saving tagged refs ...")
     with open(refs_output_path, "w", encoding="utf-8") as fout:
-        fout.write("\n".join(input_dict["refs"]))
+        for repo, repo_refs in sorted(input_dict["refs"].items()):
+            for ref in sorted(repo_refs):
```

Comment:
`repo_refs` is already sorted, and calling sorted may actually ruin the previous sorting which is not necessary the same, eg `v1.0` comes after `v1.0-rc1`

Reply:
Aaaaaah thanks for spotting that :+1:

Reply:
did you forget to add the change or ?

Reply:
huh I reverted it by mistake